TestJob Polling Should Consider Response Status Code

befishe opened the pull request:
We're seeing a lot of test job queries from the ISE client, despite the fact that the response from the B/E is NotFound; Polling should take status code into account.

befishe committed:
TestJob Polling Should Consider Response Status Code

We're seeing a lot of test job queries from the ISE client, despite the
fact that the response from the B/E is NotFound; Polling should take
status code into account.

JAL9302 commented:
Code looks fine but it seems all it is doing is changing button enabled-ness? How does this prevent continual polling on not found test jobs?

befishe committed:
Stop polling on InternalServerError too

Including InternalServerError in the list of http status codes to cause
end of polling.

befishe commented:
@Diastro will add InternalServerError as per offline discussion!
@jodoglevy if you look at the else clause (that I didn't modify) in the checkTestJob method, you'll see that we have this: `refreshTimer.Enabled = true;`
This is how we continue with the polling - by not doing this, we essentially stop polling.

eamonoreilly commented:
Change looks good - merging into master.
